### PR TITLE
#204 rejoining rooms after logging in

### LINKED ISSUE #204: Keep list of joined rooms of a player.
**Is your feature request related to a problem? Please describe.**
Keep list of room members, and rejoin players upon logging in.

**Describe the solution you'd like**
We should have `rooms` collection with below schema:
```
{
    "id"
    "members": [...usernames]
}
```

Upon logging in, we should make query for rooms where user is on the list of members, and then join newly logged in user to each of found rooms.

Frontend should listen for JOIN event, and whenever the username of join event is the same as the user's, then it checks whether there's already known room with certain ID, and if not, then it creates new record in store and selects the new room.

**Describe alternatives you've considered**
N/A

**User story**
1. I log in
2. I'm being added to every room that I'm a member or

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by michaelKurowski**
**Reference to related ticket:**
resolves #204 
**Description of changes:**

### CODE REVIEW

**Comment by michaelKurowski** [comment]
TODO:
- upon JOIN event from client, find a room by ID, update its members list.
- upon CONNECT, find rooms by username in `members` propery, and for each found room send a `JOIN` event to user

**Comment by michaelKurowski** [comment]
Ready for CR

File: src/server/ws-routes/controllers/Room.js [hunk 1]
```diff
@@ -33,25 +38,39 @@ class Room {
...
+		
 		const {roomId} = data
 		if (_.isEmpty(roomId)) return
 		const username = socket.request.user.username
-		
-		socket.join(roomId, () => {
-			const response = new JoinResponse(username, roomId)
-			MessageModel.find({ roomId }, 'author text date', (err, messages) => {
-				if (err) return logger.error(err)
+		RoomModel.findOne({id: roomId}, 'id members')
```
Comment (MarcinGrzeszczak) [inline src/server/ws-routes/controllers/Room.js]: Queries without exec() method are not promises. Please add exec()

File: src/server/ws-routes/controllers/Room.js [hunk 2]
```diff
@@ -33,25 +38,39 @@ class Room {
...
+					return
+				}
+				this.create({invitedUsersIndexes: [], roomId}, socket, connections)
+			})
+			.then(() => {
+				if (!data.roomId) return
+				return this.listMembers(data, socket)
+			})
+			.then(() => MessageModel.find({ roomId }, 'author text date'))
+			.then(messages => {
```
Comment (MarcinGrzeszczak) [inline src/server/ws-routes/controllers/Room.js]: Please make this code more beauty. Too much intends.

File: src/server/ws-routes/controllers/Room.js [hunk 3]
```diff
@@ -131,23 +160,15 @@ class Room {
 
 	[EVENT_TYPES.LIST_MEMBERS](data, socket) {
 		const roomId = data.roomId
-		const room = socket.nsp.in(roomId)
-		getRoomClients(room)
-			.then(clients => {
-				const usernames = _.map(clients, socketId => 
-					getUsername(room.connected[socketId]))
-				const response = new ListMembersResponse(usernames, roomId)
+		return RoomModel.findOne({id: roomId}, 'members')
```
Comment (MarcinGrzeszczak) [inline src/server/ws-routes/controllers/Room.js]: Another query without exec. Please do the same as above.

**Pull request merged**